fe/fe-core/src/main/java/com/starrocks/catalog/TempExternalTable.java

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    private void parseFiles() throws DdlException {
+        try {
+            HdfsUtil.parseFile(path, new BrokerDesc(properties), fileStatuses);
+        } catch (UserException e) {
+            throw new DdlException("failed to parse files: " + e.getMessage());
+        }
+    }
+
+    private PGetFileSchemaRequest getGetFileSchemaRequest(List<TBrokerFileStatus> filelist) throws TException {
+        TBrokerScanRangeParams params = new TBrokerScanRangeParams();
+        params.setUse_broker(false);
+        params.setSrc_slot_ids(new ArrayList<>());
+        params.setProperties(properties);
+
+        try {
+            THdfsProperties hdfsProperties = new THdfsProperties();
+            HdfsUtil.getTProperties(filelist.get(0).path, new BrokerDesc(properties), hdfsProperties);
+            params.setHdfs_properties(hdfsProperties);
+        } catch (UserException e) {
+            throw new TException("failed to parse files: " + e.getMessage());
+        }
+
+        TBrokerScanRange brokerScanRange = new TBrokerScanRange();
+        brokerScanRange.setParams(params);
+        brokerScanRange.setBroker_addresses(Lists.newArrayList());
+
+        for (int i = 0; i < filelist.size(); ++i) {
+            TBrokerRangeDesc rangeDesc = new TBrokerRangeDesc();
+            rangeDesc.setFile_type(TFileType.FILE_BROKER);
+            rangeDesc.setFormat_type(TFileFormatType.FORMAT_PARQUET);
+            rangeDesc.setPath(filelist.get(i).path);
+            rangeDesc.setSplittable(filelist.get(i).isSplitable);
+            rangeDesc.setStart_offset(0);
+            rangeDesc.setFile_size(filelist.get(i).size);
+            rangeDesc.setSize(filelist.get(i).size);
+            rangeDesc.setNum_of_columns_from_file(0);
+            rangeDesc.setColumns_from_path(new ArrayList<>());
+            brokerScanRange.addToRanges(rangeDesc);
+        }
+
+        TScanRange scanRange = new TScanRange();
+        scanRange.setBroker_scan_range(brokerScanRange);
+
+        final TGetFileSchemaRequest tRequest = new TGetFileSchemaRequest();
+        tRequest.setScan_range(scanRange);
+
+        final PGetFileSchemaRequest pRequest = new PGetFileSchemaRequest();
+        pRequest.setRequest(tRequest);
+        return pRequest;
+    }
+
+    private List<Column> getFileSchema() throws DdlException {
+        TNetworkAddress address;
+        try {
+            List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().getBackendIds(true);
+            if (backendIds.isEmpty()) {
+                throw new DdlException("Failed to send proxy request. No alive backends");
+            }
+            Collections.shuffle(backendIds);
+            Backend be = GlobalStateMgr.getCurrentSystemInfo().getBackend(backendIds.get(0));
+            address = new TNetworkAddress(be.getHost(), be.getBrpcPort());
+
+            // TODO(fw): more format support.
+            PGetFileSchemaRequest request = getGetFileSchemaRequest(fileStatuses);
```

Comment:
check `fileStatuses` is not empty

Reply:
https://github.com/StarRocks/starrocks/blob/a67f0811baf713acadbd4fc578690b35d70f9eb4/fe/fe-core/src/main/java/com/starrocks/catalog/TempExternalTable.java#L142-L144